components/proxy/src/test/java/com/hotels/styx/startup/AdminServerSetUpTest.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2013-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.startup;
+
+import com.google.common.collect.ImmutableMap;
+import com.hotels.styx.StyxConfig;
+import com.hotels.styx.admin.AdminServerConfig;
+import com.hotels.styx.api.HttpHandler2;
+import com.hotels.styx.api.configuration.Configuration;
+import com.hotels.styx.api.service.spi.StyxService;
+import com.hotels.styx.client.applications.BackendService;
+import com.hotels.styx.infrastructure.Registry;
+import com.hotels.styx.server.HttpConnectorConfig;
+import com.hotels.styx.server.HttpServer;
+import com.hotels.styx.server.HttpsConnectorConfig;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeoutException;
+
+import static com.hotels.styx.api.support.HostAndPorts.freePort;
+import static java.util.Collections.emptyList;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class AdminServerSetUpTest {
```

Comment:
I would get rid of this test, because
- The AdminServerSetup has only one static method that initialises the admin server builder.
- This really is an integration test, not an unit test. We ought to have the integration tests in e2e suite.
- I doubt this adds any more value over the e2e suite we already have in place.